lib/logflare/backends.ex

```diff
@@ -169,25 +181,86 @@ defmodule Logflare.Backends do
   @doc """
   Adds log events to the source event buffer.
   The ingestion pipeline then pulls from the buffer and dispatches log events to the correct backends.
+
+  TODO: Perform syncronous parsing and validation of log event params.
+
+  Once this function returns `:ok`, the events get dispatched to respective backend adaptor portions of the pipeline to be further processed.
   """
   @type log_param :: map()
   @spec ingest_logs([log_param()], Source.t()) :: :ok
-  def ingest_logs(log_events, source) do
-    via = via_source(source, :buffer)
-    Buffer.add_many(MemoryBuffer, via, log_events)
+  def ingest_logs(event_params, source) do
+    :ok = Source.Supervisor.ensure_started(source)
+    {log_events, errors} = split_valid_events(source, event_params)
+    count = Enum.count(log_events)
+    increment_counters(source, count)
+    maybe_broadcast_and_route(source, log_events)
+    RecentLogsServer.push(source, log_events)
+    dispatch_to_backends(source, log_events)
+    if Enum.empty?(errors), do: {:ok, count}, else: {:error, errors}
+  end
+
+  defp split_valid_events(source, event_params) do
+    event_params
+    |> Enum.reduce({[], []}, fn param, {events, errors} ->
+      le =
+        param
+        |> case do
+          %LogEvent{} = le ->
+            le
+
+          param ->
+            LogEvent.make(param, %{source: source})
+        end
+        |> Logs.maybe_mark_le_dropped_by_lql()
+        |> LogEvent.apply_custom_event_message()
+
+      cond do
+        le.drop ->
+          do_telemetry(:drop, le)
+          {events, errors}
+
+        le.valid == false ->
```

Comment:
```suggestion
        not le.valid ->
```

Doesn't that make more sense?